aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -217,26 +217,21 @@ at::Tensor scalar_to_tensor_default_dtype(
   }
 }
 
-// TLDR: Don't call with `use_default_dtype` true -- this is only necessary to support the partial
+// TLDR: Don't call `wrapped_scalar_tensor_default_dtype` -- this function is only necessary to support the partial
 // type-promotion that torch.where supports.  Once torch.where fully supports type promotion, we
 // won't need this function.
 //
 // Longer explanation:
-// `use_default_dtype` is a bit of a hack because torch.where doesn't support type promotion, but
+// `wrapped_scalar_tensor_default_dtype` is a bit of a hack because torch.where doesn't support type promotion, but
 // does support `torch.where(tensor, scalar1, scalar2)` with default scalar types.  The trickiness is we
 // usually convert double scalars to doubles, and `set_wrapped_number` defines type promotion priority
 // as being below tensor types rather than as the default dtype (perhaps we should?).  This wouldn't matter
 // if we just supported type normal type promotion on torch.where, however.
-Tensor wrapped_scalar_tensor(
+Tensor wrapped_scalar_tensor_default_dtype(
```

Comment:
`torch.where`'s usage of `wrapped_scalar_tensor` is weird, as per the comments. Before, the implementation of `wrapped_scalar_tensor` in this file took in an extra flag that said whether or not to use the special `torch.where` behavior. I just folded that into its own function.